include/boost/fusion/container/set/set.hpp

```diff
@@ -97,39 +97,20 @@ namespace boost { namespace fusion
 
         template <typename Sequence>
         BOOST_FUSION_GPU_ENABLED
-        set(Sequence const& rhs)
-            : data(rhs) {}
-
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
-        template <typename Sequence>
-        BOOST_FUSION_GPU_ENABLED
         set(Sequence&& rhs)
             : data(std::forward<Sequence>(rhs)) {}
-#endif
 
         BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(typename detail::call_param<T>::type ...args)
             : data(args...) {}
 
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
         template <typename ...U>
         BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(U&& ...args)
```

Comment:
> > Yes. During constructing (or copying, moving, ...) nested container, outer container's constexpr ctor tries to instantiate inner container's ctor including implicit conversion ctor whcih has SFINAE, like enable_if>::type, even if such ctor is not eligible.

> I believe this was due to the implementation of is_convertible, and not specifically caused by the mixture of SFINAE and constexpr. If you remove is_convertible from the expression in the prior bug report, the code compiles. This does make SFINAE difficult, because the expression cannot invoke anything that might cause an unexpected evaluation/instantiation.

Ahhhhh! Yes! You're right. I misunderstood. OK, I'll add `is_sequence` checking.